lumicks/pylake/nb_widgets/fd_selector.py

```diff
@@ -3,14 +3,14 @@
 import time
 
 
-class FdRangeSelectorWidget:
-    def __init__(self, fd_curve, axes=None, show=True):
-        self._axes = axes if axes else plt.axes(label=f"lk_widget_{time.time()}")
-        self.fd_curve = fd_curve
-        self.time = self.fd_curve.f.timestamps
+class SliceRangeSelectorWidget:
+    def __init__(self, channel_slice, axes=None, show=True):
+        self._axes = axes if axes else plt.axes(label=f"lk_slice_widget_{time.time()}")
+        self.slice = channel_slice
+        self.time = self.slice.timestamps
         self.connection_id = None
         self.current_range = []
-        self.ranges = []
+        self._ranges = []
```

Comment:
Has anything changed here? If the meaning has changed, a more descriptive name would be preferred.

Reply:
There's an attribute that grants access to a copy of ranges now (named `ranges`). The issue with allowing direct access to it is that the user can edit them and this might cause issues for the widget if the users changes them to some invalid format. This way, we communicate that `_ranges` is private and that once they edit the actual source data, all bets are off.

API-wise, nothing changes other than that they get a copy rather than a reference of these values.